driver-core/src/main/com/mongodb/client/model/expressions/StringExpression.java

```diff
@@ -35,13 +35,25 @@ public interface StringExpression extends Expression {
 
     StringExpression substr(IntegerExpression start, IntegerExpression length);
 
-    default StringExpression substr(int start, int length) {
+    default StringExpression substr(final int start, final int length) {
         return this.substr(of(start), of(length));
     }
 
     StringExpression substrBytes(IntegerExpression start, IntegerExpression length);
 
-    default StringExpression substrBytes(int start, int length) {
+    default StringExpression substrBytes(final int start, final int length) {
         return this.substrBytes(of(start), of(length));
     }
+
+    /**
+     * parses to an underlying int64
+     * @return
+     */
+    IntegerExpression parseInteger();
+
+    DateExpression parseDate();
+
+    DateExpression parseDate(StringExpression format);
```

Comment:
Can you clarify why we have an overload with no time zone (iirc we removed overloads like this for the inverse operation)

Reply:
It is possible that a set of strings will encode the timezone to be used within each string itself, in which case it is impossible to specify as a parameter to parseDate.

See: https://github.com/mongodb/mongo-java-driver/pull/1050/files#diff-e3444bc38b15c4823c706e26593cb36cbcf4bdb6b656c6fdb356b6f0a1f14c91R208

```
        // throws: "cannot pass in a date/time string with GMT offset together with a timezone argument"
        assertThrows(MongoCommandException.class, () ->
                assertExpression(1, of("2007-12-03T10:15:30.005+01:00")
                        .parseDate(of("+01:00"), of("%Y-%m-%dT%H:%M:%S.%L%z"))
                        .asString()));
        // therefore, to parse date strings containing UTC offsets, we need:
        assertExpression(
                Instant.parse("2007-12-03T09:15:30.005Z"),
                of("2007-12-03T10:15:30.005+01:00")
                    .parseDate(of("%Y-%m-%dT%H:%M:%S.%L%z")),
                "{'$dateFromString': {'dateString': '2007-12-03T10:15:30.005+01:00', "
                        + "'format': '%Y-%m-%dT%H:%M:%S.%L%z'}}");
```